libs/state/effects/src/lib/effects.service.ts

```diff
@@ -72,17 +67,9 @@ import { toHook, untilDestroyed } from './utils';
  */
 @Injectable()
 export class RxEffects implements OnDestroy$ {
-  constructor(
-    @Optional()
-    private readonly errorHandler: ErrorHandler | null,
-  ) {
-    inject(DestroyRef).onDestroy(() => {
-      this._hooks$.next({ destroy: true });
-      this.subscription.unsubscribe();
-    });
-  }
-
   private static nextId = 0;
+  private readonly destroyRef = inject(DestroyRef);
+  private readonly errorHandler = inject(ErrorHandler, { optional: true });
```

Comment:
isn't that a breaking change? if someone extended from `RxEffects` it can lead to issues

Reply:
Technically yes, but the recommended usage is by injecting RxEffects, not extending, do we want to consider inheritance as a use case we want to support?

Reply:
It is only a breaking change imho when `RxEffects` is instantiated with `new`.
Inheritance with instantiating via DI still works the same if I do not miss something.

Reply:
if someone extended and used `super(errorHandler)` it will cause an error after we merged your PR. That's the only scenario I can think of. 

I guess even no single person in the universe is doing that ^^

Reply:
Your commits are the hints, so be careful what you write :)